sip-core/src/main/java/de/ikor/sip/foundation/core/premiumsupport/registration/TelemetryData.java

```diff
@@ -0,0 +1,189 @@
+package de.ikor.sip.foundation.core.premiumsupport.registration;
+
+import de.ikor.sip.foundation.core.actuator.routes.AdapterRouteSummary;
+import java.net.InetAddress;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.UnknownHostException;
+import java.util.*;
+import javax.validation.constraints.Max;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotBlank;
+import lombok.AccessLevel;
+import lombok.Data;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.hibernate.validator.constraints.Length;
+import org.springframework.boot.actuate.health.Status;
+import org.springframework.core.env.Environment;
+import org.springframework.validation.annotation.Validated;
+
+/** POJO that is used to transmit adapter instance specific data to the SIP Backend. */
+@Data
+@Slf4j
+@Validated
+/** Content of adapter-platform heartbeat requests. */
+class TelemetryData {
+
+  public TelemetryData(SIPRegistrationProperties configProps, Environment environment) {
+    this.setInstanceUri(configProps.getInstanceUri(), environment);
+    this.instanceId = configProps.getInstanceId();
+    setStage(configProps.getStage(), environment);
+  }
+
+  /**
+   * This is a unique id that identifies an instance of an adapter. Its value is assigned from
+   * {@link SIPRegistrationProperties}
+   */
+  @Setter(AccessLevel.NONE)
+  private UUID instanceId;
+
+  /**
+   * The instance uri is composed of {@link #getInstanceScheme()}, {@link #getInstanceHost} and
+   * {@link #getInstancePort()}
+   */
+  private URI instanceUri;
+
+  /**
+   * The name of this adapter instance. This value must be defined in the configuration file.
+   * Maximum length of adapter name is 64.
+   */
+  @NotBlank()
+  @Length(max = 64)
+  private String adapterName;
+
+  /** The version of the SIP Framework. This value must be defined in the configuration file. */
+  private String version = "1.0.0";
+
+  /**
+   * The interval the adpater registeres at the SIP Backend. The value of register interval has to
+   * between 1000ms and 120000ms. Default value is 30000 milliseconds.
+   */
+  @Min(1000)
+  @Max(120000)
+  private Long interval = 30000L;
```

Comment:
Should be fetched from Property file (in the same way as InstanceId and InstanceUri)